cli/tests/pcluster/templates/test_cluster_stack/test_head_node_dna_json/awsbatch-headnode-hooks-semi.head-node.dna.json

```diff
@@ -0,0 +1,48 @@
+{
```

Comment:
A similar file is duplicated for every test for the dna.json.

We can add a constant dictionary in the test_cluster_stack and then every test will modify and then verify only the impacted fields.

I mean something like:
```python

DEFAULT_DNA_JSON = { "cluster": { ...} }

@pytest.mark.parametrize(
    "config_file_name, expected_head_node_dna_json_fields",
    [
     ("awsbatch-headnode-hooks-semi.yaml", {"postupdate": "https://test2.tgz", "postupdate_args": "NONE"}),
     ("slurm-headnode-hooks-full.yaml", {"postupdate": "https://test2.tgz", "postupdate_args": "NONE"}),
    ...
        
    expected_head_node_dna_json = DEFAULT_DNA_JSON["cluster"].update(expected_head_node_dna_json_fields)
```

In this way if adding new fields in the dna.json is not required to modify all the dna.json of all the tests but only the DEFAULT_DNA_JSON definition.

Reply:
That's a great point, will try to implement that for the head node tests